### DISCUSSION TITLE
Revision Updates

### DISCUSSION THREAD

**Discussion by sinclairzx81** [sinclairzx81/typebox-codegen#24] [General]
This thread is setup to notify of revision updates published to NPM

**Comment by sinclairzx81** [comment]
@xddq Hi! Hope you're doing well.

Hey, just to let you know, I just published `@sinclair/typebox-codegen@0.9.0` up to npm to match the updates in TypeBox 0.32.0. There shouldn't be any breaking changes in this revision, but have bumped a minor semver as the updates in TypeBox 0.32.0 have required  changes to be made across all code generators. 

You can read the changelog for TB 0.32.0 at the link below

https://github.com/sinclairzx81/typebox/blob/master/changelog/0.32.0.md

This update adds mapped typed and subtractive modifier TS to TB translation. As per other similar updates, these can test this on the Workbench.

[TypeBox Workbench: Mapped Types](https://sinclairzx81.github.io/typebox-workbench/?share=Ly8gLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLQovLyBUeXBlQm94IE1hcHBlZCBUeXBlcyAKLy8gLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLQoKZXhwb3J0IHR5cGUgVCA9IHsKICB4OiBzdHJpbmcsCiAgeTogbnVtYmVyLAogIHo6IGJvb2xlYW4KfQoKLy8gZXhhbXBsZXMgMSAtIGtleXMgYW5kIHByb3BlcnRpZXMKZXhwb3J0IHR5cGUgQSA9IHsgW0sgaW4ga2V5b2YgVF06IEsgfQpleHBvcnQgdHlwZSBCID0geyBbSyBpbiBrZXlvZiBUXTogVFtLXSB9CgovLyBleGFtcGxlcyAyIC0gYWRkIGFuZCByZW1vdmUgbW9kaWZpZXJzCmV4cG9ydCB0eXBlIEMgPSB7IFtLIGluIGtleW9mIFRdPzogSyB9CmV4cG9ydCB0eXBlIEQgPSB7IFtLIGluIGtleW9mIFRdPzogVFtLXSB9CmV4cG9ydCB0eXBlIEUgPSB7IFtLIGluIGtleW9mIENdLT86IENbS10gfQpleHBvcnQgdHlwZSBGID0geyBbSyBpbiBrZXlvZiBEXS0%2FOiBEW0tdIH0KCi8vIGV4YW1wbGVzIDMgLSBjb25kaXRpb25hbCBtYXAga2V5IGFuZCBwcm9wZXJ0eSB0eXBlcwpleHBvcnQgdHlwZSBHID0geyAKICBbSyBpbiBrZXlvZiBUXTogCiAgICBLIGV4dGVuZHMgJ3gnID8gJ2ZvbycgOgogICAgSyBleHRlbmRzICd5JyA%2FICdiYXInIDoKICAgIEsgZXh0ZW5kcyAneicgPyAnYmF6JyA6CiAgICBuZXZlcgp9CmV4cG9ydCB0eXBlIEggPSB7IAogIFtLIGluIGtleW9mIFRdOiAKICAgIFRbS10gZXh0ZW5kcyBzdHJpbmcgPyAnZm9vJyA6CiAgICBUW0tdIGV4dGVuZHMgbnVtYmVyID8gJ2JhcicgOgogICAgVFtLXSBleHRlbmRzIGJvb2xlYW4gPyAnYmF6JyA6CiAgICBuZXZlcgp9CgovLyBleGFtcGxlcyA0IC0gbmVzdGVkIG1hcHBpbmcKZXhwb3J0IHR5cGUgSSA9IHsKICBbWCBpbiAnYScgfCAnYiddIDogewogICAgW1kgaW4gJ2MnIHwgJ2QnXSA6IHsKICAgICAgW1ogaW4gJ2UnIHwgJ2YnXSA6IFtYLCBZLCBaXSAKICAgIH0gCiAgfSAKfQ%3D%3D)

Let me know if you run into any issues upgrading to this version downstream.
Cheers!
S

**Reply by xddq** [reply]
@sinclairzx81 Hey! Hope you are also doing good!

Great stuff you are going at with typebox! Will take a look at the changes in the next several weeks and then consider upgrading ts2typebox and schema2typebox.

Cheers! :]
